scout/server/utils.py

```diff
@@ -126,7 +126,7 @@ def safe_redirect_back(request: LocalProxy, link: Optional[str] = None) -> Respo
     if referrer:
         parsed_referrer = urlparse(referrer)
         if parsed_referrer.netloc == request.host:
-            return redirect(link or referrer)
+            return redirect(link or referrer, code=307)
```

Comment:
Allow POST request redirects. Otherwise forms are cleared on submit, validation and subsequent fail redirect. It feels natural to be able to keep editing after a small error. 😁